scapy/arch/linux.py

```diff
@@ -408,102 +408,55 @@ def _flush_fd(fd):
-        self.outs.setsockopt(socket.SOL_SOCKET, socket.SO_SNDBUF, 2**30)
-        self.promisc = conf.promisc if promisc is None else promisc
-        if self.promisc:
-            if iface is None:
-                self.iff = get_if_list()
-            elif isinstance(iface, list):
-                self.iff = iface
-            else:
-                self.iff = [iface]
-            for i in self.iff:
-                set_promisc(self.ins, i)
-
-    def close(self):
-        if self.closed:
-            return
-        self.closed = 1
-        if self.promisc:
-            for i in self.iff:
-                set_promisc(self.ins, i, 0)
-        SuperSocket.close(self)
-
-    def recv_raw(self, x=MTU):
-        """Receives a packet, then returns a tuple containing (cls, pkt_data, time)"""  # noqa: E501
-        pkt, sa_ll = self.ins.recvfrom(x)
-        if sa_ll[2] == socket.PACKET_OUTGOING:
-            return None, None, None
-        if sa_ll[3] in conf.l2types:
-            cls = conf.l2types[sa_ll[3]]
-            self.lvl = 2
-        elif sa_ll[1] in conf.l3types:
-            cls = conf.l3types[sa_ll[1]]
-            self.lvl = 3
-        else:
-            cls = conf.default_l2
-            warning("Unable to guess type (interface=%s protocol=%#x family=%i). Using %s", sa_ll[0], sa_ll[1], sa_ll[3], cls.name)  # noqa: E501
-            self.lvl = 2
-        ts = get_last_packet_timestamp(self.ins)
-        return cls, pkt, ts
-
-    def recv(self, x=MTU):
-        pkt = SuperSocket.recv(self, x)
-        if pkt and self.lvl == 2:
-            pkt = pkt.payload
-        return pkt
-
-    def send(self, x):
-        iff, a, gw = x.route()
-        if iff is None:
-            iff = conf.iface
-        sdto = (iff, self.type)
-        self.outs.bind(sdto)
-        sn = self.outs.getsockname()
-        ll = lambda x: x
-        if type(x) in conf.l3types:
-            sdto = (iff, conf.l3types[type(x)])
-        if sn[3] in conf.l2types:
-            ll = lambda x: conf.l2types[sn[3]]() / x
-        sx = raw(ll(x))
-        x.sent_time = time.time()
-        try:
-            self.outs.sendto(sx, sdto)
-        except socket.error as msg:
-            if msg[0] == 22 and len(sx) < conf.min_pkt_size:
-                self.outs.send(sx + b"\x00" * (conf.min_pkt_size - len(sx)))
-            elif conf.auto_fragment and msg[0] == 90:
-                for p in x.fragment():
-                    self.outs.sendto(raw(ll(p)), sdto)
-            else:
-                raise
+def get_iface_mode(iface):
+    """Return the interface mode"""
+    p = subprocess.Popen(["iwconfig", iface], stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT)
+    output, err = p.communicate()
+    match = re.search(br"mode:([a-zA-Z]*)", output.lower())
+    if match:
+        return plain_str(match.group(1))
+    return "unknown"
+
+
+def set_iface_monitor(iface, monitor):
+    mode = get_iface_mode(iface)
```

Comment:
Could you add a docstring?

Reply:
Done. I don't think those objects are tested